modules/simulation/JsEvManager/index.js

```diff
@@ -310,257 +203,274 @@ function simulation_loop(mod) {
+function modify_charging_session(mod, args) {
+  if (mod === undefined) {
+    evlog.warning('Already received data, but framework is not ready yet');
+    return;
+  }
 
-    const c = mod.registeredCmds[cmdname];
-    if (c === undefined || args.length != c.numargs) {
-      // Ignoreing unknown command / command with wrong parameter count
-      evlog.warning(`Ignoring command ${cmdname} with ${args.length} arguments`);
-    } else {
-      compiledCommands.push({ cmd: c.cmd, args, exec: c.exec });
-    }
+  if (!mod.enabled) {
+    evlog.warning('Simulation disabled, cannot execute charging simulation.');
+    return;
   }
-  return compiledCommands;
+
+  const str = args.value;
+
+  mod.simCommands = parseSimCommands(mod, str);
+  mod.loopCurrentCommand = -1;
+  if (next_command(mod)) mod.executionActive = true;
+}
+
+function registerCmd(mod, name, numargs, execfunction) {
+  mod.registeredCmds[name] = { cmd: name, numargs, exec: execfunction };
 }
 
 // Register all available sim commands
 function registerAllCmds(mod) {
   mod.registeredCmds = [];
 
-  registerCmd(mod, 'cp', 1, (mod, c) => {
-    mod.simdata_setting.cp_voltage = c.args[0];
-    return true;
-  });
-
   registerCmd(mod, 'sleep', 1, (mod, c) => {
-    if (c.timeLeft === undefined) c.timeLeft = c.args[0] * 4 + 1;
-    return (!(c.timeLeft-- > 0));
+    if (c.timeLeft === undefined) {
+      c.timeLeft = c.args[0] * 4 + 1;
+    }
+    c.timeLeft -= 1;
+    return (!(c.timeLeft > 0));
   });
 
-  registerCmd(mod, 'iec_wait_pwr_ready', 0, (mod, c) => {
+  registerCmd(mod, 'iec_wait_pwr_ready', 0, (mod) => {
     mod.state = 'pluggedin';
-    if (mod.simulation_feedback === undefined) return false;
-    if (mod.simulation_feedback.evse_pwm_running && dutyCycleToAmps(mod.simulation_feedback.pwm_duty_cycle) > 0) {
+    if (mod.pwm_duty_cycle > 7.0 && mod.pwm_duty_cycle < 97.0) {
       return true;
     }
     return false;
   });
 
-  registerCmd(mod, 'iso_wait_pwm_is_running', 0, (mod, c) => {
+  registerCmd(mod, 'iso_wait_pwm_is_running', 0, (mod) => {
     mod.state = 'pluggedin';
-    if (mod.simulation_feedback === undefined) return false;
-    if (mod.simulation_feedback.evse_pwm_running && mod.simulation_feedback.pwm_duty_cycle > 0.04
-      && mod.simulation_feedback.pwm_duty_cycle < 0.97) {
+    // AC ISO can also start with nominal dutycycle
+    if (mod.pwm_duty_cycle > 4.0 && mod.pwm_duty_cycle < 97.0) {
       return true;
     }
     return false;
   });
 
   registerCmd(mod, 'draw_power_regulated', 2, (mod, c) => {
-    mod.maxCurrent = c.args[0];
-    mod.numPhases = c.args[1];
+    mod.uses.ev_board_support.call.set_ac_max_current({ current: c.args[0] });
+    if (c.args[1] === 3) mod.uses.ev_board_support.call.set_three_phases({ three_phases: true });
+    else mod.uses.ev_board_support.call.set_three_phases({ three_phases: false });
     mod.state = 'charging_regulated';
     return true;
   });
 
+  // Todo(sl). Check power_fixed again
```

Comment:
Is that still relevant?

Reply:
Is still relevant, but for a later PR. By moving the simulation parts from the CarSimulator to the YetiSimulator, the implementation of the power_fixed cmd is more difficult.